quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/players/TCPPlayer.java

```diff
@@ -32,16 +42,41 @@ public void illegalMovePlayed(String msg) {
 
     @Override
     public void trigger(GameEvent event) {
-
+        if (event instanceof TurnActionEvent) {
+            TurnAction turnAction = ((TurnActionEvent) event).getAction();
+            String nextPlayer = ((TurnActionEvent) event).getNextPlayer();
+            boolean isGameEnded = ((TurnActionEvent) event).isGameEnded();
+            List<Position> currentPlayerMoves = ((TurnActionEvent) event).getCurrentPlayerMoves();
+            EndTurnEvent endTurnEvent = new EndTurnEvent(turnAction, nextPlayer,
+                    isGameEnded, currentPlayerMoves);
+            gameWebSocket.endTurn(gameId, endTurnEvent);
+
+        }
     }
 
     @Override
     public void setBoard(ReadOnlyPhysicalBoard board) {
+        this.board = board;
+    }
 
+    @Override
+    public String getPlayerName() {
+        return this.name;
     }
 
     @Override
     public TurnAction play() {
-        return null;
+        while (gameWebSocket.getLastTurnAction() == null) {
+            try {
+                Thread.sleep(100);
```

Comment:
@Guyeise1 want to talk about this

it will take a lot of time for the AI training